### ISSUE TITLE
False positive "Possibly uninitialized value" when unpacking tuples

### ISSUE THREAD

**Issue by kklimonda** [intellij-rust/intellij-rust#8427]
## Environment

* **IntelliJ Rust plugin version:** 0.4.162.4321-213
* **Rust toolchain version:** rustc 1.60.0-nightly (1bd4fdc94 2022-01-12)
* **IDE name and version:** CLion 2021.3.2
* **Operating system:** macOS 11.6

## Problem description

The following code generates false positve messages "Possibly uninitialized value" for both tuple assignment, and reassigning later.

```rust
fn main() {
  let x;
  let _y;
  
  (x, _y) = (1, 2); // Warning about x and _y being possibly uninitialized
  
  let _z = x; // Warning about x being possibly uninitialized
}
```

## Steps to reproduce

Copy&paste code into CLion

**Comment by ccleve** [comment]
Ugly workaround:

```
let result = (1, 2);
x = result.0;
_y = result.1;

```
It would be very helpful if we could get this bug fixed, though.
